src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.dev.cinema.service;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isPresent()
+                && isValid(userFromDb.get().getPassword(), password, userFromDb.get().getSalt())) {
+            return userFromDb.get();
+        }
+        throw new AuthenticationException("Incorrect email " + email + " or password");
+    }
+
+    @Override
+    public User register(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isPresent()) {
+            throw new AuthenticationException("This email " + email + " is already used");
+        }
```

Comment:
let's remove this check, we will validate input later on in controller + you throw AuthenticationException while registration - these are different procedures

Reply:
fixed